railties/lib/rails/generators/rails/app/templates/bin/yarn.tt

```diff
@@ -1,5 +1,7 @@
 APP_ROOT = File.expand_path('..', __dir__)
 Dir.chdir(APP_ROOT) do
+  # exclude bin from PATH to prevent loop.
+  ENV["PATH"] = ENV["PATH"].split(":").reject { |path| path == "bin" }.join(":")
```

Comment:
`":"` is used on Linux / Mac, but `";"` is used on Windows.  Ruby to the rescue with `File::PATH_SEPARATOR`!

Also, using `File.expand_path` will let us catch e.g. `./bin` too.

```suggestion
  ENV["PATH"] = ENV["PATH"].split(File::PATH_SEPARATOR).reject do |path|
    File.expand_path(path) == __dir__
  end.join(File::PATH_SEPARATOR)
```

Reply:
Thanks :)